src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
I don't think the behavior here is correct. MIR, as stored in crate metadata, should be universal and not depend on how the crate was compiled, but how crates which use it will be compiled. Unless we don't want incremental recompilation to allow toggling between debug mode and release mode that easily?
cc @nikomatsakis